cloudify_types/cloudify_types/component/component.py

```diff
@@ -376,20 +376,28 @@ def delete_deployment(self):
 
         ctx.logger.info('Delete component\'s "{0}" deployment'
                         .format(self.deployment_id))
-        self._http_client_wrapper('deployments',
-                                  'delete',
-                                  dict(deployment_id=self.deployment_id))
 
-        ctx.logger.info("Waiting for component's deployment delete.")
-        poll_result = poll_with_timeout(
-            lambda: deployment_id_exists(self.client, self.deployment_id),
-            timeout=self.timeout,
-            expected_result=False)
+        poll_result = True
+        if not deployment_id_exists(self.client, self.deployment_id):
+            # Could happen in case that deployment failed to install
+            ctx.logger.warn('Didn\'t find component\'s "{0}" deployment,'
+                            'so nothing to do and moving on.'
+                            .format(self.deployment_id))
+        else:
+            self._http_client_wrapper('deployments',
+                                      'delete',
+                                      dict(deployment_id=self.deployment_id))
 
-        ctx.logger.info("Little wait internal cleanup services.")
+            ctx.logger.info("Waiting for component's deployment delete.")
+            poll_result = poll_with_timeout(
+                lambda: deployment_id_exists(self.client, self.deployment_id),
+                timeout=self.timeout,
+                expected_result=False)
+
+        ctx.logger.debug("Little wait internal cleanup services.")
         time.sleep(POLLING_INTERVAL)
-        ctx.logger.info("Wait for stop all system workflows.")
 
+        ctx.logger.info("Wait for stop all system workflows.")
```

Comment:
"Waiting for all system workflows to stop/finish..."
In any case, I would change this, and the former log to be printed with every poll if it's possible with the retry number?
Just better UX instead of just keeping the user waiting with this message without seeing any actual progress...